### PR TITLE
dev-cpp/benchmark: add 1.9.0 and adoption

### PR DESCRIPTION
**Pull request by beatussum**
Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-cpp/benchmark/benchmark-1.8.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+SRC_URI="https://github.com/google/benchmark/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="Apache-2.0"
+SLOT="0/$(ver_cut 1)"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~loong ~ppc ~ppc64 ~riscv ~x86"
+IUSE="default-libcxx doc libpfm lto +exceptions test +tools"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="
+	default-libcxx? ( sys-libs/libcxx[${MULTILIB_USEDEP}] )
```
Comment (juippis) [inline dev-cpp/benchmark/benchmark-1.8.5.ebuild:21]: Could you rename this flag to simply `libcxx` so it's consistent with other packages in gentoo tree?

File: dev-cpp/benchmark/benchmark-1.8.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+pkg_setup() {
+	use tools && python-single-r1_pkg_setup
+}
+
+multilib_src_configure() {
+	local mycmakeargs=(
+		-DBENCHMARK_ENABLE_DOXYGEN="$(usex doc)"
+		-DBENCHMARK_ENABLE_EXCEPTIONS="$(usex exceptions)"
+		-DBENCHMARK_ENABLE_GTEST_TESTS="$(usex test)"
+		-DBENCHMARK_ENABLE_LTO="$(usex lto)"
```
Comment (juippis) [inline dev-cpp/benchmark/benchmark-1.8.5.ebuild:52]: Could you check what exactly `+lto` does in the build system? If it **just** adds `-flto` to your `*FLAGS`, then we can get rid of the use flag in here.

File: dev-cpp/benchmark/metadata.xml [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+	</maintainer>
+	<maintainer type="project" proxied="proxy">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<use>
+		<flag name="default-libcxx">Use <pkg>sys-libs/libcxx</pkg> instead of libstdc++ as the default stdlib</flag>
+		<flag name="libpfm">Enable <pkg>dev-libs/libpfm</pkg> support</flag>
+		<flag name="exceptions">Enable the use of exceptions in the benchmark library</flag>
+		<flag name="tools">Install some additional Python tools</flag>
```
Comment (juippis) [inline dev-cpp/benchmark/metadata.xml:16]: Please sort.

File: dev-cpp/benchmark/metadata.xml [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
+	</use>
+	<use lang="fr">
+		<flag name="default-libcxx">Utiliser <pkg>sys-libs/libcxx</pkg> à la place de libstdc++ comme bibliothèque standard</flag>
+		<flag name="libpfm">Activer le support de <pkg>dev-libs/libpfm</pkg></flag>
+		<flag name="exceptions">Utiliser les exceptions inclues dans la bibliothèque</flag>
+		<flag name="tools">Installer des outils Python additionnels</flag>
```
Comment (juippis) [inline dev-cpp/benchmark/metadata.xml:22]: (Here too)

File: dev-cpp/benchmark/benchmark-1.8.5.ebuild (hunk 2 above)
Comment (beatussum) [reply dev-cpp/benchmark/benchmark-1.8.5.ebuild:52]: It seems that `BENCHMARK_ENABLE_LTO` do not only add `-flto` ([here](https://github.com/google/benchmark/blob/c19cfee61e136effb05a7fc8a037b0db3b13bd4c/CMakeLists.txt#L163-L181) and [here](https://github.com/google/benchmark/blob/c19cfee61e136effb05a7fc8a037b0db3b13bd4c/CMakeLists.txt#L249-L264)).

File: dev-cpp/benchmark/benchmark-1.8.5.ebuild (hunk 1 above)
Comment (beatussum) [reply dev-cpp/benchmark/benchmark-1.8.5.ebuild]: Done!

File: dev-cpp/benchmark/metadata.xml (hunk 3 above)
Comment (beatussum) [reply dev-cpp/benchmark/metadata.xml]: Done!

File: dev-cpp/benchmark/metadata.xml (hunk 4 above)
Comment (beatussum) [reply dev-cpp/benchmark/metadata.xml]: Done!

File: dev-cpp/benchmark/benchmark-1.9.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+DEPEND="
+	libcxx? ( sys-libs/libcxx[${MULTILIB_USEDEP}] )
+	libpfm? ( dev-libs/libpfm:= )
+"
+
+BDEPEND="
+	>=dev-build/cmake-3.10
+	test? ( dev-cpp/gtest[${MULTILIB_USEDEP}] )
+"
```
Comment (juippis) [inline dev-cpp/benchmark/benchmark-1.9.0.ebuild:28]: You need doxygen dep with `+doc` use flag here.

File: dev-cpp/benchmark/benchmark-1.9.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	cmake_src_configure
+}
+
+multilib_src_install_all() {
+	dodoc CONTRIBUTING.md
+	dodoc CONTRIBUTORS
+
+	python_domodule tools/gbench
+	python_doscript tools/compare.py
+	python_doscript tools/strip_asm.py
```
Comment (juippis) [inline dev-cpp/benchmark/benchmark-1.9.0.ebuild:70]: These are only installable with `+tools` use flag, please make this conditional.
```
 * ERROR: dev-cpp/benchmark-1.9.0::gentoo failed (install phase):
 *   No Python implementation set (EPYTHON is null).
 *
 * Call stack:
 *     ebuild.sh, line  136:  Called src_install
 *   environment, line 3606:  Called cmake-multilib_src_install
 *   environment, line 1250:  Called multilib-minimal_src_install
 *   environment, line 2491:  Called multilib_src_install_all
 *   environment, line 2953:  Called python_domodule 'tools/gbench'
 *   environment, line 3068:  Called die
 * The specific snippet of code:
 *       [[ -n ${EPYTHON} ]] || die 'No Python implementation set (EPYTHON is null).';
```

**Review by juippis — COMMENTED** [review]
Minor fixes left, I can fix them while merging but would appreciate if you did meanwhile.

**Comment by beatussum** [comment]
No problem, I'm going to fix that

File: dev-cpp/benchmark/benchmark-1.9.0.ebuild (hunk 5 above)
Comment (beatussum) [reply dev-cpp/benchmark/benchmark-1.9.0.ebuild]: Done!

File: dev-cpp/benchmark/benchmark-1.9.0.ebuild (hunk 6 above)
Comment (beatussum) [reply dev-cpp/benchmark/benchmark-1.9.0.ebuild]: Done!

**Review by juippis — APPROVED** [review]
Awesome, thanks!

**Pull request closed without merging**